### PR TITLE
Script to repair the excessive number of configs

### PR DESCRIPTION
**Pull request by kaspervd**
See https://github.com/HiSPARC/datastore/issues/23

### CODE REVIEW

File: migration/config-repair/remove_incorrect_configs.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+def remove_configs(filename, cluster, stationnumber):
+    """
+    Remove all but the first configuration row
+    """
+    
+    copyfile(filename, filename+'_backup')
+
+    with tables.open_file(filename, "a") as data:
+        config = data.get_node("/hisparc/cluster_"+cluster+"/station_"+str(stationnumber)+"/config")
```
Comment (153957) [inline migration/config-repair/remove_incorrect_configs.py]: Use string formatting:
```python
"/hisparc/cluster_{cluster}/station_{stationnumber}/config".format(cluster=cluster, stationnumber=stationnumber)
```

File: migration/config-repair/remove_incorrect_configs.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        print len(config)
+        config.remove_rows(1,len(config))
```
Comment (153957) [inline migration/config-repair/remove_incorrect_configs.py]: The end point for [`remove_rows`](http://www.pytables.org/usersguide/libref/structured_storage.html#tables.Table.remove_rows) is optional
> If None all rows after start will be removed.

File: migration/config-repair/remove_incorrect_configs.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+
+
+if __name__ == "__main__":
+    remove_configs("2017_4_15.h5", "amsterdam", 4)
```
Comment (153957) [inline migration/config-repair/remove_incorrect_configs.py]: Do not forget station 91 on 2017-4-12.

File: migration/config-repair/remove_incorrect_configs.py (hunk 2 above)
Comment (tomkooij) [inline migration/config-repair/remove_incorrect_configs.py]: I think `remove_rows(2)` should be enough.
http://www.pytables.org/usersguide/libref/structured_storage.html?highlight=remove_rows#tables.Table.remove_rows

File: migration/config-repair/remove_incorrect_configs.py (hunk 3 above)
Comment (153957) [inline migration/config-repair/remove_incorrect_configs.py]: And perhaps use the full path, or provide a more complete example of how this is to be executed.
Similar to: https://github.com/HiSPARC/datastore/blob/master/migration/migrate_gps_test_data.md#frome-has-an-older-version-of-pytables

File: migration/config-repair/remove_incorrect_configs.py (hunk 1 above)
Comment (tomkooij) [inline migration/config-repair/remove_incorrect_configs.py]: `data.config('/hisaprc/cluster_%s/station_%d/config' % (cluster, stationnumber))` is much more readable. IMHO

File: migration/config-repair/remove_incorrect_configs.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import tables
+from shutil import copyfile
+
+
+def remove_configs(filename, cluster, stationnumber):
+    """
+    Remove all but the first configuration row
+    """
+    
+    copyfile(filename, filename+'_backup')
```
Comment (tomkooij) [inline migration/config-repair/remove_incorrect_configs.py]: Don't backup in your script!

Backup. Check backup. Make sure the backup is untouchable by your script.

Then run the script.

**Review by tomkooij — CHANGES_REQUESTED** [review]
Disregard the syntax stuff, but please backup (and verify the backup) before starting the script.

**Review by 153957 — CHANGES_REQUESTED** [review]

**Comment by 153957** [comment]
Also add extra explanations to a README/docstrings:
- Why these configurations are going to be removed
- What caused it
- What is not fixed (and why), i.e. the blobs are left.

**Comment by kaspervd** [comment]
Thanks for the review. I've made the changes and am working on the README. Where in the data file is station 91? Also, should I also place the publicdb repair scripts here?

**Comment by 153957** [comment]
You can find the cluster to which 91 belongs here:
http://data.hisparc.nl/api/station/91/

Yes, also add the publicdb cleanup scripts here.

**Pull request merged**